### PR TITLE
Add docs for defining desired resource state

### PR DESCRIPTION
**Pull request by alinabuzachis**
##### SUMMARY

Add docs for defining desired resource state

##### ISSUE TYPE

- Docs Pull Request

### CODE REVIEW

File: docs/docsite/rst/desired_state_schema.rst [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+    type: object
+    properties:
+      ResourceGroupName:
+        description: The unique identifier for a group of resources. ResourceGroupName should not include any spaces.
+        type: string
+      Resources:
+        description: Dictionary of resources belonging to ResourceGroupName.
+        type: object
+        properties:
+          name:
```
Comment (gravesm) [inline docs/docsite/rst/desired_state_schema.rst:22]: I'm assuming this is JSON Schema, in which case, isn't this saying that a resource needs a property literally called `name`? I think we would want to use `patternProperties` here, wouldn't we?

File: docs/docsite/rst/desired_state_schema.rst [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+            description: The unique identifier for a resource. It is a hash map with the name as the key."
```
Comment (gravesm) [inline docs/docsite/rst/desired_state_schema.rst:23]:
```suggestion
            description: A locally unique identifier for a resource. It is a hash map with the name as the key."
```
I would clarify here that this is locally unique and shouldn't be confused with the cloud provider's unique identifier for the resource.

File: docs/docsite/rst/desired_state_schema.rst (hunk 1 above)
Comment (alinabuzachis) [reply docs/docsite/rst/desired_state_schema.rst:22]: It should be the YAML definition schema.

Comment (gravesm) [reply docs/docsite/rst/desired_state_schema.rst:22]: The only YAML schema I'm aware of is based on JSON schema. Is there something else?

Comment (alinabuzachis) [reply docs/docsite/rst/desired_state_schema.rst:22]: I took it from here https://asdf-standard.readthedocs.io/en/1.5.0/schemas/yaml_schema.html

Comment (alinabuzachis) [reply docs/docsite/rst/desired_state_schema.rst:22]: What I meant is something like this:
```
- ResourceGroupName: cloud_1
    Resources:
    bucket_02:
      Type: AWS::S3::Bucket
```
where that name is literally bucket_02.

**Comment by alinabuzachis** [comment]
recheck

**Comment by alinabuzachis** [comment]
recheck

**Review by gravesm — APPROVED** [review]

**Comment by alinabuzachis** [comment]
recheck

**Comment by alinabuzachis** [comment]
recheck

**Pull request merged**